### ISSUE TITLE
Package libmonolith

### ISSUE THREAD

**Issue by yonas** [Y2Z/monolith#417]
Would be nice for Rust apps, like https://github.com/freetonik/textpod, to make use of monolith without having to install a separate binary.

**Comment by ebenoist** [comment]
Agreed. A declared library would also make a WASM port more possible.

**Comment by snshn** [comment]
Hello Yonas and Erik,

it's in the works, definitely will be part of 3.0, perhaps even sooner than that.  I'm currently working on making this software asynchronous, and then solidifying the API after that.  I hate when the API changes as much as the next cat does, so don't wanna do it too often, there's enough suffering in the world as is.

**Comment by greyblake** [comment]
One more here. I would love also to use monololith as a library instead of a binary.

P.S. Just want to thank the authors and contributors for this extremely useful piece of software :heart:

**Comment by erik-nilcoast** [comment]
> P.S. Just want to thank the authors and contributors for this extremely useful piece of software

Amen, I literally use it ever day. I have it in production currently :). It's exactly what I needed.

**Comment by Ohrest88** [comment]
I would also really like to use monolith, which is written in Rust, in my Flutter application via the flutter_rust_bridge

**Comment by snshn** [comment]
As of now, there're two functions that can be imported and used directly from the crate:
- https://docs.rs/monolith/2.10.0/monolith/core/fn.create_monolithic_document.html
- https://docs.rs/monolith/2.10.0/monolith/core/fn.create_monolithic_document_from_data.html

`>>> Here be dragons <<<`

This is super raw, certain to change a lot, but it should work directly (around the executable).  I've added `[lib]` to Cargo.toml, now _in theory_, libmonolith.so (or .dll) can be built and used via bindings from other languages.

If there're any problems, suggestions, etc — please let me know (in this issue or feel free to open a new one).

@freetonik, you might find it useful for textpod

**Comment by snshn** [comment]
> Agreed. A declared library would also make a WASM port more possible.

@ebenoist it's already been done some years ago, https://github.com/rhysd/monolith-of-web + https://github.com/rhysd/monolith
